### PR TITLE
switch from macos-12 pin to macos-latest

### PR DESCRIPTION
**Pull request by atravitz**
closes https://github.com/OpenFreeEnergy/gufe/issues/313

minimum supported macos should be 14.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -33,10 +33,11 @@ jobs:
           - "3.10"
           - "3.11"
           - "3.12"
+          - "3.13"
```
Comment (IAlibay) [inline .github/workflows/ci.yaml:36]: Looks like we're not 3.13 compatible yet (because of OpenMM 8.2)

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (atravitz) [reply .github/workflows/ci.yaml:36]: Yep! Just wanted to check. Removing now.

**Review by IAlibay — APPROVED** [review]
please do raise an issue re: macos-13 (or reference in an already existing one)

**Comment by atravitz** [comment]
> please do raise an issue re: macos-13 (or reference in an already existing one)

https://github.com/OpenFreeEnergy/gufe/issues/425

**Pull request merged**